common/core/desktop/tests/unit/kmx/kmx.cpp

```diff
@@ -306,6 +315,13 @@ int run_test(const km::kbp::path & source, const km::kbp::path & compiled) {
   km_kbp_context_item *citems = nullptr;
   try_status(km_kbp_context_items_from_utf16(context.c_str(), &citems));
   try_status(km_kbp_context_set(km_kbp_state_context(test_state), citems));
+
+  // Make a copy of the setup context for the test
+  std::vector<km_kbp_context_item> test_context;
+  for(km_kbp_context_item *ci = citems; ci->type != KM_KBP_CT_END; ci++) {
+    test_context.emplace_back(*ci);
+  }
+
```

Comment:
Do we have to add `KM_KBP_CT_END` to `test_context`?

Reply:
No, because we are using a vector here (so we know length) and we are manipulating the end of `test_context`, we don't actually want the `KM_KBP_CT_END` terminator.

Reply:
That makes sense. However, I'm surprised that line 375 works where we check for `KM_KBP_CT_END` but not for the end of the vector.